d1.go

```diff
@@ -0,0 +1,197 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"github.com/goccy/go-json"
+	"net/http"
+	"time"
+)
+
+var (
+	ErrMissingDatabaseID = fmt.Errorf("required missing database ID")
+)
+
+type D1Database struct {
+	Name      string     `json:"name"`
+	NumTables int        `json:"num_tables"`
+	UUID      string     `json:"uuid"`
+	Version   string     `json:"version"`
+	CreatedAt *time.Time `json:"created_at"`
+	FileSize  int64      `json:"file_size"`
+}
+
+type ListD1Parameters struct {
+	Name string `url:"name,omitempty"`
+	ResultInfo
+}
+
+type ListD1Response struct {
+	Result []D1Database `json:"result"`
+	Response
+	ResultInfo `json:"result_info"`
+}
+
+type CreateD1Parameters struct {
+	Name string `json:"name"`
+}
+
+type D1DatabaseResponse struct {
+	Result D1Database `json:"result"`
+	Response
+}
+
+type QueryD1Parameters struct {
+	DatabaseID string   `json:"database_id"`
+	SQL        string   `json:"sql"`
+	Parameters []string `json:"params"`
+}
+
+type D1DatabaseMetadata struct {
+	ChangedDB   bool    `json:"changed_db"`
```

Comment:
https://github.com/cloudflare/cloudflare-go/blob/master/docs/conventions.md#booleans

```suggestion
	ChangedDB   *bool    `json:"changed_db"`
```